packages/mui-material/src/Dialog/Dialog.js

```diff
@@ -282,56 +285,101 @@ const Dialog = React.forwardRef(function Dialog(inProps, ref) {
     return { titleId: ariaLabelledby };
   }, [ariaLabelledby]);
 
+  const backwardCompatibleSlots = {
+    transition: TransitionComponent,
+    ...slots,
+  };
+
+  const backwardCompatibleSlotProps = {
+    transition: TransitionProps,
+    paper: PaperProps,
+    backdrop: BackdropProps,
+    ...slotProps,
+  };
+
+  const externalForwardedProps = {
+    slots: backwardCompatibleSlots,
+    slotProps: backwardCompatibleSlotProps,
+  };
+
+  const [RootSlot, rootSlotProps] = useSlot('root', {
```

Comment:
```suggestion
  const [RootSlot, rootSlotProps] = useSlot('root', {
    shouldForwardComponentProp: true,
```

If the elementType is a styled component of another styled component, this prop should be set to true, so that `slotProps.root.component` does not replace the inner styled component.

Reply:
got it added here https://github.com/mui/material-ui/pull/44792/commits/51ca92bdb08fadb1a08ce75ac2ed1b75f02fa445. 

>If the elementType is a styled component of another styled component, this prop should be set to true, so that slotProps.root.component does not replace the inner styled component.

Does that mean, `shouldForwardComponent:true` is missing here? given `TableSortLabelRoot` is of another styled component (ButtonBase)

https://github.com/sai6855/material-ui/blob/51ca92bdb08fadb1a08ce75ac2ed1b75f02fa445/packages/mui-material/src/TableSortLabel/TableSortLabel.js#L132-L133